ios-sdk/src/MapViewController.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  MapViewController.swift
+//  ios-sdk
+//
+//  Created by Matt Smollinger on 11/21/16.
+//  Copyright © 2016 Mapzen. All rights reserved.
+//
+
+
+import UIKit
+import TangramMap
+
+public class MapViewController: TGMapViewController {
+
+    var currentLocationGem: TGMapMarkerId?
+
+    public var showCurrentLocation: Bool = false {
+        didSet {
+            if showCurrentLocation == true {
+                if currentLocationGem != nil {
+                    //Already showing the current location gem
+                    return;
+                }
+                print("Queuing scene update for current location")
+                let point = TGGeoPoint(longitude: -122.44880676269531, latitude: 37.76155490343394)
+                let marker = markerAdd()
+                //TODO: Update once scene updates are properly synchronous - { style: ux-location-gem-overlay, interactive: true, sprite: ux-current-location, size: 36px, collide: false }
+
+                markerSetStyling(marker, styling: "{ style: 'points', color: 'white', size: [25px, 25px], order:500, collide: false }")
+                markerSetPoint(marker, coordinates: point)
+                markerSetVisible(marker, visible: true)
+                animateToPosition(point, withDuration: 2.0)
+                animateToZoomLevel(15, withDuration: 2.0)
+                currentLocationGem = marker
+            } else {
+                guard let marker = currentLocationGem else {
+                    // Not showing current location
+                    return;
+                }
+                markerRemove(marker)
```

Comment:
Toggling visibility rather than adding/removing the marker seems like good practice for optimization of this and possibly other marker overlays to come.

If we're not going to update the behavior in this change set we should make an issue to update it later so we don't forget.